### PR TITLE
Remove python3 dependencies implied by ament_python build type.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
From https://github.com/ros/rosdistro/pull/15328 there is no longer a rosdep key for `python3`.
All ament_python packages have injected build_depends on Debian packages python3-all, dh-python, and python3-setuptools.
The Debian package builder also adds python runtime dependencies as it sees fit via the `${python3:Depends}` helper.

I think there's a long term danger of too-closely relying on assumed dependencies, particularly for platform-specific builds, but for consistency with other ament_python packages I have removed these here.

If we want to leave these dependencies untouched in the source package.xml, they'll need to be patched out at release time since there is no longer a rosdep key `python3`.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Couldn't the dependency on `python3-setuptools` be removed too?

**Comment by nuclearsandwich** [comment]
> Couldn't the dependency on python3-setuptools be removed too?

tl;dr: If we want to provide the minimum necessary information in package.xml yes. If we want package.xml to provide an accurate and relatively complete view of a package's dependencies, or consistently enter python `exec_depend`s even if in this case, the dependency is already required at build time we should leave it.

---

Yes. It could and I'd be open to removing it, I left it to talk about whether or not there's value re-reflecting dependencies in the package.xml or if we're going to refer folks to the setup.py for the canonical dependencies for ament_python packages. But I also am not sure whether or not the ability to remove the dependency key from the package.xml applies generally to python libraries. Since python3-setuptools is a build-depend of every ament_python package. It should be there and detected by dh-python as a runtime dependency as well. But if another ament_python package depends on python3-foobar, I think python3-foobar will need to appear as a build dependency in order to be present on the system and be injected. This is analogous to depending on a `libfoo-dev` package at buildtime and having dh-shlibdeps provide the corresponding runtime dependency.

**Comment by nuclearsandwich** [comment]
Do we want CI for this change? Because python3 is a manually installed dependency rather than part of the current ament build order, I don't believe this change will affect the build.

**Comment by dirk-thomas** [comment]
> Do we want CI for this change

Not necessary for the reason you described.
